homeassistant/components/beewi_smartclim/sensor.py

```diff
@@ -1,85 +1,136 @@
 """Platform for beewi_smartclim integration."""
+
 from __future__ import annotations
 
-from beewi_smartclim import BeewiSmartClimPoller  # pylint: disable=import-error
-import voluptuous as vol
+from bleak.backends.device import BLEDevice
+from smartclim_ble import BeeWiSmartClimAdvertisement
 
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataProcessor,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+    PassiveBluetoothProcessorCoordinator,
+    PassiveBluetoothProcessorEntity,
+)
 from homeassistant.components.sensor import (
-    PLATFORM_SCHEMA,
     SensorDeviceClass,
     SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
 )
-from homeassistant.const import CONF_MAC, CONF_NAME, PERCENTAGE, UnitOfTemperature
+from homeassistant.const import ATTR_NAME, PERCENTAGE, UnitOfTemperature, UnitOfTime
 from homeassistant.core import HomeAssistant
-import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import DEVICE_CLASS_NAME, DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-# Default values
-DEFAULT_NAME = "BeeWi SmartClim"
-
-# Sensor config
-SENSOR_TYPES = [
-    [SensorDeviceClass.TEMPERATURE, "Temperature", UnitOfTemperature.CELSIUS],
-    [SensorDeviceClass.HUMIDITY, "Humidity", PERCENTAGE],
-    [SensorDeviceClass.BATTERY, "Battery", PERCENTAGE],
-]
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_MAC): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    }
-)
-
 
-def setup_platform(
```

Comment:
Please implement `async_setup_platform` and import the config from configuration.yaml via a config flow import step to a config entry.

See previous PRs that added a config flow to existing integrations.